File: code3-6.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    //入力
    int N, W;
    cin >> N >> W;
    vector<int> a(N);
    for(int i = 0; i < N; ++i) cin >> a[i];

    //bitは2^N通りの部分集合全体を動きます
    bool exist = false;
    for(int bit = 0; bit < (1 << N); ++bit)
    {
        int sum = 0; //部分集合に含まれる要素の和
        for (int i = 0; i < N; ++i){
            //i番目の要素a[i]が部分集合に含まれているかどうか
            if (bit & (1 << i)) {
                sum += a[i];
            }
        }

        //sumがWと一致するか
        if (sum == W) exist = true;
    }

    if (exist) cout << "Yes" << endl;
    else cout << "No" << endl;
}

File: code3-6.py
n, w = map(int, input().split())
a = []
for i in range(n):
    ai = int(input())
    a.append(ai)

exist = False
for bit in range((1 << n)):
    sum = 0
    for i in range(n):
        if bit & (1 << i):
            sum += a[i]

    if sum == w:
        exist = True

if exist:
    print("Yes")
else:
    print("No")